core/src/main/java/org/mule/runtime/core/privileged/processor/MessageProcessors.java

```diff
@@ -276,9 +280,74 @@ public static BaseEventContext newChildContext(CoreEvent event, Optional<Compone
     return internalProcessWithChildContext(event, processor, true, childContext);
   }
 
+  /**
+   * Process a {@link Processor} using a child {@link EventContext}. This is useful if it is necessary to perform processing in a
+   * scope and handle an empty result or error locally rather than complete the response for the whole Flow.
+   *
+   * @param event the event to process.
+   * @param processor the processor to process.
+   * @param componentLocation
+   * @return the result of processing processor.
+   * @throws MuleException
+   */
+  public static CoreEvent processWithChildContextBlocking(CoreEvent event, Processor processor,
+                                                          Optional<ComponentLocation> componentLocation)
+      throws MuleException {
+    return internalProcessWithChildContextBlocking(event, processor, true,
+                                                   child(((BaseEventContext) event.getContext()), componentLocation));
+  }
+
+
+  /**
+   * Process a {@link Processor} using a child {@link EventContext}. This is useful if it is necessary to perform processing in a
+   * scope and handle an empty result or error locally rather than complete the response for the whole Flow.
+   * <p>
+   * The {@link FlowExceptionHandler} configured on {@link MessageProcessorChain} or {@link FlowConstruct} will be used to handle
+   * any errors that occur.
+   *
+   * @param event the event to process.
+   * @param processor the processor to process.
+   * @param componentLocation
+   * @param exceptionHandler used to handle {@link Exception}'s.
+   * @return the result of processing processor.
+   * @throws MuleException
+   */
+  public static CoreEvent processWithChildContextBlocking(CoreEvent event, Processor processor,
+                                                          Optional<ComponentLocation> componentLocation,
+                                                          FlowExceptionHandler exceptionHandler)
+      throws MuleException {
+    return internalProcessWithChildContextBlocking(event, processor, true,
+                                                   child(((BaseEventContext) event.getContext()), componentLocation,
+                                                         exceptionHandler));
+  }
+
+  private static CoreEvent internalProcessWithChildContextBlocking(CoreEvent event, Processor processor,
+                                                                   boolean completeParentIfEmpty, BaseEventContext child)
+      throws MuleException {
+    final Publisher<CoreEvent> childResponsePublisher = child.getResponsePublisher();
+
+    CoreEvent result;
+    try {
+      result = processor.process(quickCopy(child, event));
+      completeSuccessIfNeeded().accept(result);
+    } catch (Throwable e) {
+      try {
+        result = Mono.from(childResponsePublisher).block();
+      } catch (Throwable t) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Catch Exception instead of Throwable. ([squid:S1181](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1181))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7683&issues=AWjyMA29otIXnmsUSXGf&open=AWjyMA29otIXnmsUSXGf)

[ISSUE_KEY]: AWjyMA29otIXnmsUSXGf